docs/reference/sql/functions/date-time.asciidoc

```diff
@@ -496,6 +496,53 @@ include-tagged::{sql-specs}/docs/docs.csv-spec[dateTimeParse3]
 --------------------------------------------------
 ====
 
+[[sql-functions-datetime-timeparse]]
+==== `TIME_PARSE`
+
+.Synopsis:
+[source, sql]
+--------------------------------------------------
+TIME_PARSE(
+    string_exp, <1>
+    string_exp) <2>
+--------------------------------------------------
+
+*Input*:
+
+<1> time expression as a string
+<2> parsing pattern
+
+*Output*: time
+
+*Description*: Returns a time by parsing the 1st argument using the format specified in the 2nd argument. The parsing
+format pattern used is the one from
+https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/time/format/DateTimeFormatter.html[`java.time.format.DateTimeFormatter`].
+If any of the two arguments is `null` or an empty string `null` is returned.
+
+[NOTE]
+If timezone is not specified in the time string expression and the parsing pattern, the resulting `time` will
+be in `UTC` timezone.
+
+[NOTE]
+If the parsing pattern contains date units (e.g. 'dd/MM/uuuu', 'dd-MM HH:mm:ss', etc.) an error is returned
+as the function needs to return a value of `time` type which will contain only time.
+
+[source, sql]
+--------------------------------------------------
+include-tagged::{sql-specs}/docs/docs.csv-spec[timeParse1]
+--------------------------------------------------
+
+[source, sql]
+--------------------------------------------------
+include-tagged::{sql-specs}/docs/docs.csv-spec[timeParse2]
+--------------------------------------------------
+
+[source, sql]
```

Comment:
Please include it in a NOTE similarly to: https://github.com/elastic/elasticsearch/pull/56158/files#diff-b91ca24147aa7c0d4371fa129d3e51bdR487